### PR TITLE
Add scenario (docs) for copying & transforming event streams (for versioning…

### PR DESCRIPTION
**Pull request by jokokko**
…, data modification)

### CODE REVIEW

File: src/Marten.Testing/ScenarioCopyAndReplaceStream.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	{
+		public ScenarioCopyAndReplaceStream()
+		{
+			StoreOptions(options =>
+			{
+				options.Events.StreamIdentity = StreamIdentity.AsString;
+			});
+		}
+
+		[Fact]
```
Comment (mysticmind) [inline src/Marten.Testing/ScenarioCopyAndReplaceStream.cs]: If this a sample for docs, can we remove/comment the `[Fact]` so that it does not get run as a unit test.

Comment (jokokko) [reply src/Marten.Testing/ScenarioCopyAndReplaceStream.cs]: @mysticmind: Certainly can be removed. Comes with the downside that should there be behavioural breaking changes (in this case something in the execution path leading to an exception, which is not likely :), the sample breaks & goes unnoticed.

Comment (mysticmind) [reply src/Marten.Testing/ScenarioCopyAndReplaceStream.cs]: @jokokko okay, make sense :-)

**Comment by mysticmind** [comment]
@jokokko quick comment, could you please fix the tab spacing in code changes to use 4 tab spaces?

**Comment by oskardudycz** [comment]
@jokokko this sample is interesting and something that we don't have covered. I was thinking about for quite some time. Is this something that you applied also practically in your project?

I have few concerns that we'd need to diccuss.

From my practice, my common scenario for working with event streams was:
1. Create Stream, publish some events.
2. Create projection (inline, or custom) that is updated during publishing events from streams.

Then while working on aggregates and getting actual aggregate stream - use online aggegation so `AggregateStream` method. (so this is how I was working with write model).

While doing queries I was using projection. So Inline projections, as they are snapshots of the write model state - I was using for the write model validation, etc. Custom projections - for the read model.

So having those end usage scenarios, let's consider how this sample would help, or what might be missing.

1. If we create copy of the stream, with new Id then we will get new record in our read model. How will we know that it's just a new "proper" item? How will we distinguish that we should not use old one? Imho we should publish "delete event" for the old stream. Having that for custom projection we could use `DeleteEvent` to remove the old record. For inline projection we don't have yet automated mechanism for deleting projection item, but it can be tricked (let's not get into details right now how...). So Delete Event should `more or less` should solve the case for read model.

2. Write model. This is the harder part. Currently we don't have the way to mark the stream as "deleted", or "obsolete". At least none built in way. We could think about some ["tombstoning"](https://github.com/JasperFx/marten/issues/996) but still, no out of the box solution. But having "delete event" suggested in the previous point we could implement custom "soft delete".

So to sum up, imho we should add to this sample also publishing delete event for the old stream. Having that it would make possible to remove the record from projection and also mark the stream as "deleted"/"inactive", so make soft delete with custom logic.

Imho this will make this sample more complete and more helpful for the end usage. Also maybe we could show mor real life sample - like bank account, rather than slaying monsters? :)

@jokokko what are your thoughts?

**Comment by mysticmind** [comment]
@jokokko and @oskardudycz are we planning to add further docs or we are good with this PR and add any docs further as another PR?

**Comment by oskardudycz** [comment]
@mysticmind that's more question to @jokokko. From my perspective it would be nice to extend this scenario with handling DeleteEvent.

**Comment by mysticmind** [comment]
@oskardudycz shall we go ahead to merge the current changes and raise a separate issue for your comments?

**Comment by mysticmind** [comment]
@oskardudycz since you merged this, I have set the milestone for this back to 3.5.0
